kv/driver.go

```diff
@@ -74,7 +77,37 @@ func NewRedisDriver(log *zap.Logger, key string, cfgPlugin Configurer, tracer *s
 		RouteByLatency:   d.cfg.RouteByLatency,
 		RouteRandomly:    d.cfg.RouteRandomly,
 		MasterName:       d.cfg.MasterName,
-	})
+	}
+
+	if d.cfg.TLSConfig.CaFile != "" {
+		tlsConfig := &tls.Config{
+			MinVersion: tls.VersionTLS12,
+		}
+
+		rootCAs, sysCertErr := x509.SystemCertPool()
+		if sysCertErr != nil {
+			rootCAs = x509.NewCertPool()
+			d.log.Warn("unable to load system certificate pool, using empty pool", zap.Error(sysCertErr))
+		}
+
+		if _, crtExistErr := os.Stat(d.cfg.TLSConfig.CaFile); crtExistErr != nil {
+			return nil, errors.E(op, crtExistErr)
+		}
+
+		bytes, crtReadErr := os.ReadFile(d.cfg.TLSConfig.CaFile)
+		if crtReadErr != nil {
+			return nil, errors.E(op, crtReadErr)
+		}
+
+		if !rootCAs.AppendCertsFromPEM(bytes) {
+			return nil, errors.E(op, errors.Errorf("failed to parse certificates from PEM file '%s'. Please ensure the file contains valid PEM-encoded certificates", d.cfg.TLSConfig.CaFile))
+		}
+
+		tlsConfig.RootCAs = rootCAs
+		redisOptions.TLSConfig = tlsConfig
+	}
```

Comment:
It is better to move TLS configuration to the separate package

Reply:
sorry i am new to go, when you say new package do you mean in another repo or simply in another directory?? :sweat_smile:

Reply:
Sure, np, I'll help you 😃 
Just a new file called `tlsconfig.go` or similar :)